packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -408,11 +409,46 @@ export interface SdkModelPropertyTypeBase extends DecoratedType {
   crossLanguageDefinitionId: string;
 }
 
+export interface SerializationOptions {
+  json?: JsonSerializationOptions;
+  xml?: XmlSerializationOptions;
+  multipart?: MultipartOptions;
+}
+
+/**
+ * For Json serialization.
+ * The name will come from explicit setting of `@encodedName("application/json", "NAME")`.
+ */
+export interface JsonSerializationOptions {
+  name: string;
+}
+
+/**
+ * For Xml serialization.
+ * The `name`/`itemsName` will come from explicit setting of `@encodedName("application/xml", "NAME")` or `@xml.Name("NAME")`.
+ * Other properties come from `@xml.attribute`, `@xml.ns`, `@xml.unwrapped`.
+ * The `itemsName` and `itemsNs` are used for array items.
+ */
+export interface XmlSerializationOptions {
```

Comment:
For boolean like `attribute` and `unwrapped`, what would it mean if it is undefined? Is it an "attribute" or not an "attribute"? I'd hope we had some clarity in TCGC.

Reply:
for xml serialization, only property could have this flag. so for model, set it to false is meaningless for model.

Reply:
So, for property, it would always have a true/false on `attribute`?
And what about `unwrapped`?

If possible, have it in document.

Reply:
> for xml serialization, only property could have this flag. so for model, set it to false is meaningless for model.

BTW, why a model has a `name` in XML? I don't think model has a `name` in JSON. Is XML something different?

Reply:
xml will use model name as the tag name. `Container` `EnumerationResults` all from the model name. refer the spec: https://github.com/Azure/azure-rest-api-specs/blob/a26af9acb30387724eff64229ba7f3259600681a/specification/storage/data-plane/Microsoft.BlobStorage/stable/2025-01-05/blob.json#L10507.